odl/util/utility.py

```diff
@@ -163,6 +185,145 @@ def is_complex_floating_dtype(dtype):
     return np.issubsctype(dtype, np.complexfloating)
 
 
+def default_dtype(impl, field):
```

Comment:
Actually, the code I posted is wrong since it can return `Rn`, for example, which is a function and does not have a `default_dtype` method. Instead we must hard-code
```python
dspc_type = CudaFn if impl == 'cuda' else Fn
dtype = dspc_type.default_dtype(fspace.field)
```